compiler/parser/Parser.y

```diff
@@ -1960,10 +1962,18 @@ tv_bndrs :: { [LHsTyVarBndr GhcPs] }
          | {- empty -}                  { [] }
 
 tv_bndr :: { LHsTyVarBndr GhcPs }
-        : tyvar                         { sL1 $1 (UserTyVar $1) }
-        | '(' tyvar '::' kind ')'       {% ams (sLL $1 $>  (KindedTyVar $2 $4))
-                                               [mop $1,mu AnnDcolon $3
-                                               ,mcp $5] }
+        : tyvar                         { sL1 $1 (UserTyVar $1 Specified) }
+        | '{' tyvar '}'                 {% ams (sLL $1 $>  (UserTyVar $2 Inferred))
+                                               [ moc $1
+                                               , mcc $3] }
+        | '(' tyvar '::' kind ')'       {% ams (sLL $1 $>  (KindedTyVar $2 $4 Specified))
+                                               [ mop $1
+                                               , mu AnnDcolon $3
+                                               , mcp $5] }
+        | '{' tyvar '::' kind '}'       {% ams (sLL $1 $>  (KindedTyVar $2 $4 Inferred))
+                                               [ moc $1
+                                               , mu AnnDcolon $3
+                                               , mcc $5] }
```

Comment:
Noted. I'm going to defer addressing this until I've gone through some of these other changes, because I don't understand happy nearly as well as I understand how to do some of these other things.

Reply:
Neither do I. But when I had shift/reduce conflicts once, it wasn't nearly as hard to fix as I thought it would be. Regardless, deferring this problem makes good sense.